File: mylibrary/src/main/java/com/xiaolu/mylibrary/net/RxManager.java
package com.xiaolu.mylibrary.net;

import java.util.HashMap;
import java.util.Map;
import java.util.Set;

import io.reactivex.disposables.CompositeDisposable;
import io.reactivex.disposables.Disposable;

/**
 * ================================================
 *
 * @ProjectName: MyLibraryTool
 * @Package: com.xiaolu.mylibrary.net
 * @ClassName: RxApiManager
 * @Description: 取消网络请求
 * @Author: xiaol
 * @CreateDate: 2020/11/3 15:53
 * @UpdateUser: xiaol
 * @UpdateDate: 2020/11/3 15:53
 * @UpdateRemark: 更新说明
 * @Version: 1.0
 * ================================================
 */
public class RxManager {
    private static RxManager instance = null;
    private Map<String, CompositeDisposable> map;

    public static RxManager getInstance() {
        if (instance == null) {
            synchronized (RxManager.class) {
                if (instance == null) {
                    instance = new RxManager();
                }
            }
        }
        return instance;
    }

    private RxManager() {
        map = new HashMap<>();
    }

    public void add(String key, Disposable disposable) {
        Set<String> keySet = map.keySet();
        if (keySet.contains(key)) {
            CompositeDisposable compositeDisposable = map.get(key);
            compositeDisposable.add(disposable);
        } else {
            CompositeDisposable compositeDisposable = new CompositeDisposable();
            compositeDisposable.add(disposable);
            map.put(key, compositeDisposable);
        }
    }

    public void clear(String key) {
        Set<String> keySet = map.keySet();
        if (keySet.contains(key)) {
            CompositeDisposable compositeDisposable = map.get(key);
            compositeDisposable.clear();
            map.remove(key);
        }
    }
}

File: mylibraryKotlin/src/main/java/com/xiaolu/mylibrarykotlin/net/RxManager.kt
package com.xiaolu.mylibrarykotlin.net

import io.reactivex.disposables.CompositeDisposable
import io.reactivex.disposables.Disposable
import java.util.*

/**
 * ================================================
 *
 * @ProjectName: MyLibraryTool
 * @Package: com.xiaolu.mylibrary.net
 * @ClassName: RxApiManager
 * @Description: java类作用描述
 * @Author: xiaol
 * @CreateDate: 2020/11/3 15:53
 * @UpdateUser: xiaol
 * @UpdateDate: 2020/11/3 15:53
 * @UpdateRemark: 更新说明
 * @Version: 1.0
 * ================================================
 */
class RxManager private constructor() {

    private val map: MutableMap<String, CompositeDisposable>

    fun add(key: String, disposable: Disposable?) {
        val keySet: Set<String> = map.keys
        if (keySet.contains(key)) {
            val compositeDisposable = map[key]
            compositeDisposable!!.add(disposable!!)
        } else {
            val compositeDisposable = CompositeDisposable()
            compositeDisposable.add(disposable!!)
            map[key] = compositeDisposable
        }
    }

    fun clear(key: String?) {
        val keySet: Set<String> = map.keys
        if (keySet.contains(key)) {
            val compositeDisposable = map[key]
            compositeDisposable!!.clear()
            map.remove(key)
        }
    }

    companion object {
        var instance: RxManager? = null
            get() {
                if (field == null) {
                    synchronized(RxManager::class.java) {
                        if (field == null) {
                            field = RxManager()
                        }
                    }
                }
                return field
            }
    }

    init {
        map = HashMap()
    }
}